lib/topology_docker_openswitch/openswitch.py

```diff
@@ -282,6 +278,66 @@ def notify_post_build(self):
         super(OpenSwitchNode, self).notify_post_build()
         self._setup_system()
 
+    # This function dumps the last "LINES_TO_DUMP" lines from the docker
+    # daemon logs
+    # Docker daemon logs can be gathered by different means depending on
+    # the host OS
+    # For example:
+    #    Ubuntu - /var/log/upstart/docker.log
+    #    CentOS - /var/log/daemon.log | grep docker
+    #    Boot2Docker - /var/log/docker.log
+    #    Debian GNU/Linux - /var/log/daemon.log
+    #    Fedora - journalctl -u docker.service
+    #    Red Hat Enterprise Linux Server - /var/log/messages | grep docker
+    #    OpenSuSE - journalctl -u docker.service
+    # For now we'll cater to Ubuntu & CentOS cases alone.
+    def dump_docker_log_file(self):
+        lines_to_dump = 100
+        docker_log_file = ""
+        docker_filter = ""
+
+        platform_system = str(system()).lower()
+        platform_version = str(version()).lower()
+
+        print("############## Docker logs info ##############")
+
+        if "linux" in platform_system:
+            if "ubuntu" in platform_version:
+                docker_log_file = "/var/log/upstart/docker.log"
+            elif "centos" in platform_version:
+                docker_log_file = "/var/log/daemon.log"
+                docker_filter = "docker"
+            else:
+                print("dumpDockerLogFile: Unknown platform")
+                return
+        else:
+            print("dumpDockerLogFile: Unknown platform")
+            return
+
+        # if path.isfile(docker_log_file) and access(docker_log_file, R_OK):
+        try:
+            output = check_output(
+                'sudo -n cat {}'.format(docker_log_file), shell=True
```

Comment:
This won't work where the user needs to provide a password to use sudo, an exception will be raised here and log collection will stop.